sqlglot/parser.py

```diff
@@ -3947,6 +3953,16 @@ def _parse_decode(self) -> t.Optional[exp.Decode | exp.Case]:
 
         return exp.Case(ifs=ifs, default=expressions[-1] if len(expressions) % 2 == 1 else None)
 
+    def _parse_encode(self) -> exp.Encode:
+        args = self._parse_csv(self._parse_conjunction)
+
+        if len(args) < 2:
+            return self.expression(
+                exp.Encode, this=seq_get(args, 0), charset=exp.Literal.string("utf-8")
+            )
+
+        return self.expression(exp.Encode, this=seq_get(args, 0), charset=seq_get(args, 1))
```

Comment:
Ditto about duplication: this is almost the same as `_parse_decode`. If we decide to keep these, we should dry them up and have them receive the expression type as a parameter in order to reuse the logic for both `Encode` and `Decode`.

Reply:
Refactored this a bit.  Hope this is closer to what you had in mind.